### PR TITLE
[CONTENT] Reshaping: Converge into the layout of `crate-clients-tools`

### PR DESCRIPTION
**Pull request by amotl**
## About

Based on GH-29, this patch converges the recently added content into roughly the same layout/shape we curated at `crate-clients-tools`, also using the same labels for the same things. See https://cratedb.com/docs/clients/.

## Preview
https://cratedb-guide--34.org.readthedocs.build/

## Details

Properly order **main section pillars**. Doing it like this, it also has a pleasant visual apperance, without too much flapping.

- Reference Architectures
- Application Domains
- Integrations

That's the layout tree of corresponding **2nd-level categories**. It may be refined on a subsequent iteration by input from others who are interested. For example, see @marijaselakovic's suggestion at GH-10.

- Application Domains
  - Time Series Data
  - Machine Learning

- Integrations
  - Load and Export
  - System Metrics
  - Data Visualization
  - Business Intelligence
  - Machine Learning
  - Software Testing

## Notes

_Notes about resolving the **Integrations vs. Topics** dilemma._

While the content is in a single place only, in order to avoid redundancy, the **separate "Application Domains" section** can now be used to **provide an alternative categorization/grouping perspective**, beyond "Integrations".

In this section, you can cherry-pick specific content sections, in order to provide better guidance into corresponding topics, optionally selecting from content within "Integrations", like "Machine Learning", or alternatively adding dedicated content like "Time Series Data", which **_physically_** does not fit into "Integrations" well.

/cc @karynzv, @marijaselakovic, @hammerhead, @hlcianfagna, @ckurze

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -141,23 +206,35 @@ optimal coverage of different application domains.
...
-
 reference-architectures/index
-topic/analysis/index
-topic/timeseries/index
-topic/ml/index
-topic/testing
+
+domain/index
+integrate/index
 ```
```
Comment (amotl) [inline docs/index.md:238]: The main toc tree is much clearer now, I like it.

File: docs/index.md [hunk 2]
```diff
@@ -89,34 +89,71 @@ use-cases.
...
+:link-type: ref
+:link-alt: Managing Time Series Data with CrateDB
+:padding: 3
+:text-align: center
+:class-card: sd-pt-3
+:class-body: sd-fs-1
+:class-title: sd-fs-5
+
+{material-outlined}`query_stats;1.3em`
+:::
```
Comment (amotl) [inline docs/index.md:111]: Currenty, "Time Series Data" and "Metrics" use the same icon. That detail may want to be improved by using a different one.

On [clients-tools » Integrations](https://cratedb.com/docs/crate/clients-tools/en/latest/#id1), it is used for "System Metrics" already, so this one should be replaced by [selecting another one](https://fonts.google.com/icons).

File: docs/index.md [hunk 3]
```diff
@@ -89,34 +89,71 @@ use-cases.
...
+:class-title: sd-fs-5
+
+{material-outlined}`transform;1.3em`
+:::
+
+
+:::{grid-item-card} Metrics
+:link: metrics
+:link-type: ref
+:link-alt: CrateDB with metrics collection agents, brokers, and stores
```
Comment (amotl) [inline docs/index.md:156]: On [clients-tools » Integrations](https://cratedb.com/docs/crate/clients-tools/en/latest/#id1), the icon here is also used for "System Metrics".

File: docs/index.md (hunk 2 above)
Comment (amotl) [reply docs/index.md:111]: Per https://fonts.google.com/icons?icon.query=measure, we've selected the [Stacked Line Chart icon](https://fonts.gstatic.com/s/i/short-term/release/materialsymbolsoutlined/stacked_line_chart/default/48px.svg). It has been implemented with 51f2788e4f8. Please let us know if you would like to select a different icon for that application domain item.

image

**Review by seut — APPROVED** [review]
👍 📖

**Pull request merged**